src/test/java/SixPMComTestSuite/SixPMFeaturesTest.java

```diff
@@ -0,0 +1,91 @@
+        WebDriverWait wait = new WebDriverWait(driver,10);
+
+        WebElement itemQuantityDropDown = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemQuantityDropDownXpath)));
+        itemQuantityDropDown.click();
+        WebElement removeButtonFromDropDown = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(removeButtonFromDropDownXpath)));
+        removeButtonFromDropDown.click();
+
+        driver.quit();
+    }
+
+    @Test
+    public void addToFavTest(){
+        WebDriverWait wait = new WebDriverWait(driver,10);
+        Actions action = new Actions(driver);
+        Random random = new Random();
+
+        driver.manage().window().maximize();
+        driver.get(homePage);
+
+        WebElement accessoriesElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(accessoriesXPath)));
+        action.moveToElement(accessoriesElem).build().perform();
+
+        WebElement aviatorsButtonElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(aviatorsButtonXPath)));
+        aviatorsButtonElem.click();
+
+        List <WebElement> listOfAllItems = wait.until(ExpectedConditions.numberOfElementsToBe(By.xpath(listOfAllItemsXPath),100));
+
+        int countOfItems = listOfAllItems.size();
+        int randomItemIndex = random.nextInt(countOfItems);
+
+        List <WebElement> actItemNameElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemNameElemActual)));
+        String actualItemName = actItemNameElem.get(randomItemIndex).getText();
+        List <WebElement> actItemInfoElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemInfoElemActual)));
+        String actualItemInfo = actItemInfoElem.get(randomItemIndex).getText();
+        List <WebElement> actItemPriceElem = wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath(itemPriceElemActual)));
+        String actualItemPrice = actItemPriceElem.get(randomItemIndex).getText();
+
+        listOfAllItems.get(randomItemIndex).click();
+
+        WebElement addToBagElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(addToBagElemXPath)));
+        addToBagElem.click();
+
+        WebElement viewBugButtonElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(viewBugButtonElemXPath)));
+        viewBugButtonElem.click();
+
+        WebElement expItemNameElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemNameElemExpected)));
+        String expectedItemName = expItemNameElem.getText();
+        WebElement expItemInfoElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemInfoElemExpected)));
+        String expectedItemInfo = expItemInfoElem.getText();
+        WebElement expItemPriceElem = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(itemPriceElemExpected)));
+        String expectedItemPrice = expItemPriceElem.getText();
```

Comment:
Here no need to have all these waits as well.
You can wait once, till the popup will be loaded with product, and then get all the actual results.

Reply:
changed